dotnet/RPCClient/RPCClient.cs

```diff
@@ -21,25 +19,31 @@ public RpcClient()
         channel = connection.CreateModel();
         replyQueueName = channel.QueueDeclare().QueueName;
         consumer = new EventingBasicConsumer(channel);
+    }
 
-        props = channel.CreateBasicProperties();
+    public string Call(string message)
+    {
+        var respQueue = new BlockingCollection<string>();
```

Comment:
Am I missing the reason why the response queue (collection) can be initialised once?

Reply:
Because otherwise it won't be thread-safe. If you run the code below with the collection initialized in the constructor, the results will be a mess because of race conditions. That's why I created a new collection for each request.

            Parallel.ForEach(new[] { "30", "5", "10", "20", "15" }, (msg) =>
             {
                 var result = client.Call(msg);
                 Console.WriteLine($"Result for {msg} is {result}");
             }
            );
As we only get one result for each request, maybe we should not use BlockingCollection. The only purpose it currently serves is as a blocking mechanism, so maybe we should use Task instead.
Let me check if I can come out with more elegant solution.

Reply:
Doh, it indeed is used a synchronisation primitive and not for keeping track of state. That must be from the .NET 1.1 days (this client is over 10 years old). A one-off `ManualResetEvent` would be much clearer.

Reply:
Hi Michael,
I already updated the code to use Task, instead of BlockingCollection. Tested it locally and works like a charm. So basically, we create a Task for each call and wait for the result at the end of the method. This uses the same blocking mechanism, however Task stores only a single value (allocates less memory).
Let me know what you think.